lib/blacklight/models/scorm_package.rb

```diff
@@ -0,0 +1,89 @@
+module Blacklight
+  class ScormPackage
+    attr_accessor(:entries, :manifest)
+
+    ##
+    # Scorm packages should include this string in the <schema> tag. We
+    # downcase, and remove spaces before checking to see if a manifest contains
+    # this schema to determine whether or not it belongs to a scorm package
+    ##
+    SCORM_SCHEMA = "adlscorm".freeze
+
+    def initialize(zip_file, manifest)
+      @manifest = manifest
+      @entries = ScormPackage.get_entries zip_file, manifest
+    end
+
+    ##
+    # Returns true if a manifest is a scorm manifest file, false otherwise
+    ##
+    def self.scorm_manifest?(manifest)
+      parsed_manifest = Nokogiri::XML(manifest.get_input_stream.read)
+      schema_name = parsed_manifest.
+        xpath("//xmlns:metadata/xmlns:schema").
+        text.delete(" ").downcase
+      return schema_name == SCORM_SCHEMA
+    # NOTE we occasionally run into malformed manifest files
+    rescue Nokogiri::XML::XPath::SyntaxError
```

Comment:
Would it be worth writing a warning to `STDERR` (or to a log file) here? Silently swallowing exceptions can often lead to _extremely_ frustrating debugging experiences.